### PR TITLE
[ROCm] changes for spack

### PR DESCRIPTION
**Pull request by i-chaochen**
Thanks @afzpatel for this work and this changes is for spack using ROCm libraries from https://github.com/ROCm/tensorflow-upstream/pull/2578

@xla-rotation

### CODE REVIEW

File: third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc [hunk 1]
```diff
@@ -40,6 +40,13 @@ string RocmRoot() {
 #endif
 }
 
-string RocdlRoot() { return io::JoinPath(RocmRoot(), "amdgcn/bitcode"); }
+string RocdlRoot() {
```
Comment (beckerhe) [inline third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc:43]: Can I ask you to use the opportunity to replace this by `std::string`?

File: third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc [hunk 2]
```diff
@@ -40,6 +40,13 @@ string RocmRoot() {
...
+  if (const char* device_lib_path_env = std::getenv("HIP_DEVICE_LIB_PATH")) {
+   return device_lib_path_env;
+  }
+  else{
+   return io::JoinPath(RocmRoot(), "amdgcn/bitcode");
+  }
```
Comment (beckerhe) [inline third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc:49]: Can you run clang-format on this? I believe the internal formatter will complain about this formatting once we import it.

**Review by beckerhe — COMMENTED** [review]
Otherwise LGTM

File: third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc (hunk 1 above)
Comment (i-chaochen) [reply third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc]: good idea! thanks for the suggestion!

File: third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc (hunk 2 above)
Comment (i-chaochen) [reply third_party/tsl/tsl/platform/default/rocm_rocdl_path.cc]: sharp-eyed!

**Comment by i-chaochen** [comment]
@beckerhe PLTL

**Review by beckerhe — APPROVED** [review]

**Comment by beckerhe** [comment]
Can you rebase onto `main`? I hope that will fix the import issue.

**Comment by i-chaochen** [comment]
> Can you rebase onto `main`? I hope that will fix the import issue.

Thanks for the notice. I have rebase it to TOT.

**Comment by i-chaochen** [comment]
Hi @xla-rotation wondering why this PR is still pending?

**Comment by i-chaochen** [comment]
gentle ping @xla-rotation

**Pull request closed without merging**